### PR TITLE
[mp4frag] Add types for mp4frag

### PR DESCRIPTION
**Pull request by kensand**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Hope everything looks good :crossed_fingers:  - Thank you for reviewing!

### CODE REVIEW

File: types/mp4frag/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    segmentCount?: number;
+}
+
+declare class Mp4Frag extends Transform {
+    /**
+     * Construct a new Mp4Frag object.
+     *
+     * @throws If options.hlsPlaylistBase contains characters other than letters(a-zA-Z) and underscores(_).
+     */
+    constructor(options: Mp4FragOptions);
```
Comment (peterblazejewicz) [inline types/mp4frag/index.d.ts:80]: aren't options optional?
https://github.com/kevinGodell/mp4frag/blob/master/index.js#L49

File: types/mp4frag/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+    /**
+     * Returns the audio codec information as a <b>string</b>.
+     * Returns <b>null</b> if requested before [initialized event]{@link Mp4Frag#event:initialized}.
+     */
+    get audioCodec(): string | undefined;
```
Comment (peterblazejewicz) [inline types/mp4frag/index.d.ts:86]: the owner opted to use implicit `null`, I'd probably stick to `null` here and in other places
https://github.com/kevinGodell/mp4frag/blob/master/index.js#L84
https://github.com/kevinGodell/mp4frag/blob/master/index.js#L88

Comment (kensand) [reply types/mp4frag/index.d.ts:86]: I considered that, and opted for undefined based on [this entry in the Typescript coding guidelines](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined) - does DefinitelyTyped have superseding guidelines for that? To be clear, I'm fine making that change, but if there's other guidelines I'm not aware of I'd like to check everything else in this PR against it too.

File: types/mp4frag/index.d.ts (hunk 1 above)
Comment (kensand) [reply types/mp4frag/index.d.ts:80]: It's not explicitly said to be optional in the documentation, but from [this line](https://github.com/kevinGodell/mp4frag/blob/master/index.js#L59) it does look to handle a null options field - good catch, will include the change, thank you.

File: types/mp4frag/index.d.ts (hunk 2 above)
Comment (peterblazejewicz) [reply types/mp4frag/index.d.ts:86]: I believe this guide was for case when writing code for TS source code. Declaration files should reflect native code, `null` is a same primitive as `undefined`. The implementation owner picked 'null' as return value (type).

**Review by gabritto — APPROVED** [review]

**Comment by kensand** [comment]
Ready to merge

**Pull request merged**